src/go/util/httppattern/uri_template_test.go

```diff
@@ -1000,10 +999,136 @@ func TestUriTemplateParseError(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(fmt.Sprintf("`%s`", tc), func(t *testing.T) {
-			if Parse(tc) != nil {
+			if ParseUriTemplate(tc) != nil {
 				t.Fatalf("succeed parsing %s but expect to fail", tc)
 			}
 		})
 
 	}
 }
+
+func TestReplaceVariableFieldInUriTemplateRebuild(t *testing.T) {
```

Comment:
How about split into two test files,  one for url_template and another for parer.

Reply:
Done.